File: sdk-java/src/main/java/com/tollara/client/model/UsageEstimateResult.java
package com.tollara.client.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

/**
 * Wire + parsed result for Core usage estimate endpoints (see docs-sdk/MAIN-SDK-API-SPEC.md §2.3).
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class UsageEstimateResult {

    private boolean sufficientCredits;
    private boolean wouldExceedCap;
    private boolean wouldAllow;
    private BigDecimal estimatedCost;
    private String billingModelType;
    private String measurementType;
    private String unitLabel;
    private UsageBreakdown breakdown;
    private int estimateSchemaVersion;
    private long timestamp;

    /** HTTP status of the Core response (200, 403, 429, …). Set by the client after parsing. */
    @JsonIgnore
    private int httpStatus;
}

File: sdk-php/src/UsageEstimateResult.php
<?php

declare(strict_types=1);

namespace Tollara\ServiceSdk;

final class UsageEstimateResult
{
    public function __construct(
        public bool $sufficientCredits,
        public bool $wouldExceedCap,
        public bool $wouldAllow,
        public ?float $estimatedCost,
        public ?string $billingModelType,
        public ?string $measurementType,
        public ?string $unitLabel,
        public ?UsageBreakdown $breakdown,
        public int $estimateSchemaVersion,
        public int $timestamp,
        public int $httpStatus,
    ) {
    }

    /** @param array<string, mixed> $data */
    public static function fromArray(array $data, int $httpStatus): self
    {
        $breakdown = null;
        if (isset($data['breakdown']) && is_array($data['breakdown'])) {
            $breakdown = UsageBreakdown::fromArray($data['breakdown']);
        }
        return new self(
            sufficientCredits: (bool) ($data['sufficientCredits'] ?? false),
            wouldExceedCap: (bool) ($data['wouldExceedCap'] ?? false),
            wouldAllow: (bool) ($data['wouldAllow'] ?? false),
            estimatedCost: isset($data['estimatedCost']) ? (float) $data['estimatedCost'] : null,
            billingModelType: isset($data['billingModelType']) ? (string) $data['billingModelType'] : null,
            measurementType: isset($data['measurementType']) ? (string) $data['measurementType'] : null,
            unitLabel: isset($data['unitLabel']) ? (string) $data['unitLabel'] : null,
            breakdown: $breakdown,
            estimateSchemaVersion: (int) ($data['estimateSchemaVersion'] ?? 0),
            timestamp: (int) ($data['timestamp'] ?? 0),
            httpStatus: $httpStatus,
        );
    }
}
